File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.cs
using System;

public class Solution 
{
    public string solution(string[] id_pw, string[,] db) 
    {
        bool idpw = false;
        string result = "";
        
        int row = db.GetLength(0);
        for(int r=0; r<row; r++) {
            if(id_pw[0] == db[r,0] && id_pw[1] == db[r,1]) {
                idpw = true;
                result = "login";
            }
        }
        
        bool id = false;
        bool pw = false;
        if(!idpw) {
            for(int r=0; r<row; r++) {
                if(id_pw[0] == db[r,0]) {
                    id = true;
                    result = "wrong pw";
                }
                else if(!id) {
                    pw = true;
                    result = "fail";
                }
            }
        }
        return result;
    }
}

File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> id_pw, vector<vector<string>> db) 
{
    string result = "fail";
    bool isLogin = false;
    for(auto &d : db)
    {
        if(id_pw == d)
        {
            result = "login";
            isLogin = true;
        }
    }
    
    if(!isLogin)
    {       
        for(auto &d : db)
        {
            if(id_pw[0] == d[0]) 
            {
                if(id_pw[1] != d[1])
                {
                    result = "wrong pw";
                }
            }
        }
    }
    return result;
}